tests/common/helpers/pfc_storm.py

```diff
@@ -93,11 +118,17 @@ def deploy_pfc_gen(self):
         Deploy the pfc generation file on the fanout
         """
         if self.peer_device.os in ('eos', 'sonic'):
+            src_pfc_gen_file = "common/helpers/{}".format(self.pfc_gen_file)
             self._create_pfc_gen()
+            if self.fanout_asic_type == 'mellanox':
+                src_pfc_gen_file = f"../ansible/roles/test/files/mlnx/docker-tests-pfcgen-asic/{self.pfc_gen_file}"
             self.peer_device.copy(
-                src="common/helpers/{}".format(self.pfc_gen_file),
+                src=src_pfc_gen_file,
                 dest=self._PFC_GEN_DIR[self.peer_device.os]
                 )
+            if self.fanout_asic_type == 'mellanox':
```

Comment:
Check `os = sonic` as well?

Reply:
No need to check os type here, the os type had been checked before.